sn_client/src/wallet.rs

```diff
@@ -145,10 +142,11 @@ impl WalletClient {
     ) -> WalletResult<((NanoTokens, NanoTokens), Vec<XorName>)> {
         let verify_store = true;
         let c: Vec<_> = content_addrs.collect();
+        let mut backoff = ExponentialBackoff::default();
         let mut last_err = "No retries".to_string();
 
-        for retry in 0..DATA_PAYMENT_RETRIES {
-            trace!("Paying for storage (retry: {retry}) for: {:?}", c);
+        while let Some(delay) = backoff.next_backoff() {
```

Comment:
this will keeps looping forwever till a success or non-send error ?
could be risky of having a deadlock.
at least have the `warn` to be a `println` as well, so that user can notice a deadlock ?

Reply:
It would hit the default backoff limit, which returns None once we've taken too much time. I'll add a log though aye

Reply:
Ah, so you mean hit that default `10s` cap of the backoff ?
ok then